Only do personalisation tasks when settings defined

michaeltchapman opened the pull request:
Tie the execution of environment personalisation tasks to whether
the settings dir has been defined, so that the environment can
be used without any customisation for testing if desired.

michaeltchapman committed:
Only do personalisation tasks when settings defined

Tie the execution of environment personalisation tasks to whether
the settings dir has been defined, so that the environment can
be used without any customisation for testing if desired.

cubeek commented on roles/development/tasks/main.yml:
```diff
@@ -20,7 +20,3 @@
     - include_tasks: python.yml
     - include_tasks: vim.yml
   ignore_errors: true
```
I guess we can remove L11-L14 too then

michaeltchapman replied:
I kept 11-14 and added a fail check, then just used my_settings_dir for checks elsewhere.

cubeek commented on roles/development/tasks/python.yml:
```diff
@@ -1,7 +1,7 @@
 ---
 - name: install packages for peresonal use via pip
   pip:
-    name: 'flake8,pep8,"ipython<5",pylint'
```
ipython 5 and above doesn't work with ipdb 0.10.2. Also newer ipdb doesn't work for me on CentOS, so I had this workaround

michaeltchapman replied:
pip was throwing errors at me saying it was an invalid requirement, I'll see if I can get to the bottom of that.

cubeek commented on roles/development/tasks/bash.yml:
```diff
@@ -3,4 +3,8 @@
   copy:
     src: "{{ my_settings_dir }}/.bashrc"
     dest: "{{ ansible_env.HOME }}/.bashrc"
+  when:
```
Isn't https://github.com/cubeek/my-ansible/blob/master/roles/development/tasks/main.yml#L23 enough?

michaeltchapman committed:
Add local conf devstack template

michaeltchapman committed:
Move python3 to development

devstack will work fine without python3, but we will
wany py3 for testing, so it can be added using install_python3

michaeltchapman committed:
Ironic wip

michaeltchapman committed:
Remove MTU and add vagrant to libvirt group before stacking

michaeltchapman committed:
Fix up neutron devstack config for initial nets

Prevent the neutron initial network setup from creating
provider networks, since with the ansible-networking config
in there as provider nets it will not work having the built
in public network as a provider.

Add a new role that cleans up the intiial ironic baremetal
config and connects the two baremetal nodes to eth2/3 instead
of ovs, both of which should be vqfx ports

michaeltchapman committed:
Fix post-devstack config

Correct port modification for virtualbaremetal, add shorthands
for becoming user and admin using openrc, and redirect to /dev/null
so that the mandatory warning doesn't pollute the output captured by
ansible.

michaeltchapman committed:
Add connectivity test

Add a small script that can run on the devstack all in one
to confirm connectivity between the ports on the appliance
that will be used for virtual baremetal instances

michaeltchapman committed:
Avoid stderr on openrc source

There's a warning that prints every time openrc is
sourced for some reason, this sends it to dev null
since it's not relevant and otherwise fills up
the ansible logs

michaeltchapman committed:
Clarify initial network creation

The neutron initial networks aren't configurable to create a vlan
tenant network based off a specific provider network, so instead
of using them we only use the ironic network config. This creates
the provisioning network, but means the role must now create the
default router, tenant network and subnet.

michaeltchapman committed:
Manage and provide virtual baremetal

prior to use, the ironic nodes need to be managed and provided.
Also added a stub for booting the node, and a note for where
to find the script that creates virtual baremetal nodes from scratch
in case we need to use that instead of the built in ironic config.

michaeltchapman committed:
Separate setup, test, teardown in conntest

Separate the phases of the connection test so it's easier to debug
and add a teardown so the node is clean when devstack runs

michaeltchapman committed:
Add tenant vlan ranges, connect br-prov as phys bridge

michaeltchapman committed:
Connect controller to appliance

michaeltchapman committed:
Move provisioning network to provider

and add ansible-networking config

michaeltchapman committed:
Connect baremetal bridges to appliance ports

michaeltchapman committed:
vlan ranges require multiples

michaeltchapman committed:
Fix ansible integer syntax

michaeltchapman committed:
Ensure libvirt and default iptables rules are flushed

There are rules both saved by default and also added by the
libvirt default network that drop packets that should be forwarded
to the appliance. Flush them and keep them flushed by installing
them before devstack runs and removing everything

michaeltchapman committed:
Only flush the REJECT rules from devstack

michaeltchapman committed:
set virtualbaremetal bridges up

michaeltchapman committed:
Fix appliance initial config

michaeltchapman committed:
Put baremetal ports on correct appliance ports

michaeltchapman committed:
Put provisioning network on VLAN 2

This should prevent things from 'accidentally' working due to provisioning
being on the default vlan.

michaeltchapman committed:
Search for appliance rather than hardcoding IP

Use an arp scan to find where the appliance is on the libvirt network
after being booted by vagrant.

Also move the mgmt iface onto a bridge so we can connect it to the
provisioning network more easily later. This is needed so that booting
nodes can get pxe from provisioning and then grab their image via http
from glance, which is on the mgmt iface.

michaeltchapman committed:
Enable ironic cleaning and provide nodes after deploy

Ironic auto clean must be disabled during devstack since
the nodes are not connected to the appliance. Once they
are connected, enable cleaning and re-provide them to
go through the cleaning process as a check that the
environment is functioning correctly

michaeltchapman committed:
Add iface to bridge script

michaeltchapman committed:
Connect br-eth0 to br-prov

michaeltchapman committed:
Make management network so that we can route to services

booting ironic nodes need to be able to access the mgmt network when
they go through pxe since it's where they get their images. Instead
of attaching eth0 to br-prov, create an openstack provider network
for eth0 and connect that to the provisioning network using a router.

michaeltchapman committed:
Add mgmt bridge before devstack runs

michaeltchapman committed:
Add bridge in devstack role

michaeltchapman committed:
Use correct ovs add bridge command

michaeltchapman committed:
Remove management network

This may not be needed as long as the provisioning network isn't attached
to the default router
